docs/project-config-en.md
# project-config

## ... translating in progress ...




docs/project-config-es.md
# project-config

_Configuración a nivel proyecto, en la raíz del mismo._


### package.json
* name: Nombre del proyecto, debe seguir las reglas de package.json típico (solo caracteres válidos, sin espacios, etc.)
* version: Se mostrara en la consola de la app y te servirá para tracking. Mantener con nomenclatura [semver] (http://semver.org/), ya que se contempla mecanismo de update según versión.

### project-config.json

**Notas:**
* **No modificar `project-config.json` del _builder_**
* Indicar la primera app, si el nombre es distinto a "app", indicarlo con "firstApp"


Este es la configuración por defecto de todos los projects, de querer cambiar algún valor hacerlo en el `project-config.json` de tu project.

```javascript
{
	"ip": "127.0.0.1",
	 /* ip donde se publicaran los servidores, con 0.0.0.0 se publica en todas y se puede acceder desde la LAN */

	"compress": false,
	/* si se encuentra la librería lz-string, podes comprimir las apps. */

	"ports": {
	/* puertos de los servers */
		...
		"template": 9002
		/* de uso interno, para desarrollo de templates */
	},

	"setGitVersionOnPackage": false,
	/* actualiza el package.json con el hash del ultimo commit */

	"autoprefixer": ["ie 10","android 4","chrome 42","ff 27","ios 7","opera 19","safari 7"],
	/* genera los prefijos css (--webkit, --ms, etc) automáticamente según versiones configuradas aquí. */

  "debugZoneActive": false,
  /* debajo de la app aparecera una zona para poder debuggear con `loader.debug` and `loader.debugAdd` */
  
  "showDeviceInfo": false,
  /* muestra en el cliente informacion del device */
  
  "showSkeletor": false,
  /* mostraria el "esquelto" de la app (estilos), ver mas en loader-es.md/#skeletor */

	"contentEditable": false,
	/* con este flag podes hacer que el contenido textual sea editable, util para conocer dimenciones y demas */

	"appCache":{
		"active": false,
		"filename": "",
		/* nombre particular, si no, le pondrá automáticamente el del proyecto */
		"ext": ".cache",
		"files": ["dist/**/*"],
		/* selección de archivos a incluir */
		"options": {
			"hash": true,
			"preferOnline": false,
			"network": ["http://!*", "https://!*", "*"]
		}
	},
	/* genera un manifest.cache automáticamente con todos los archivos del proyecto */

	"app": {
		"release": true,
		/* esto mismo a nivel app, minificando y otras tareas más */

		"firstApp": "app",
		/* primera app que va a lanzar, es requerido, tiene que saber por dónde empieza el flujo */

		"site": "http://",
		/* el site se pondrá automáticamente en el footer */

		"folders": {
			"www": "www",
			"build": "build",
			"dist": "dist",
			"temp": ".tmp"
		},
		/* configuración de folders, si necesitas cambiarle el nombre, directamente desde aquí */

		"compatibilityMatrix": {
			"ie": [9.9,10.9],
			"firefox": [23.9,25.9],
			"android": [3.9,3.9],
			"chrome": [17.9,18.9],
			"chromeMobile": [3.9,3.9],
			"safari": [6.0,6.0],
			"ios": [6,6]
		},
		/* seteo de mínimo y recomendado por cada browser, menos del mínimo da error (panic) y recomienda updatear o usar otro (ver "text"), debería haber correlación con autoprefixer */

		"consoleError": [
			"Error unknown",
			"..."
		]
		/* Errores falsos para mostrar en la consola y confundir */
	},

	"cordova":{
		"active": false,
		/* si es del tipo cordova el proyecto, active debería ser true */
		"folder": "cordova/",
		"www": "cordova/www/",
		"isDevice": false,
		/* NO CAMBIAR, ni setear, lo hará automáticamente */

		"files": {
			"index": "index-cordova.html"
		},
		"platform": {
			"android": {
				"keypass": "provisoryKey1",
				"storepass": "provisoryKey1"
			}
		}
		/* parámetros según plataforma */
	},
	/* utilización de cordova para generar proyecto hibrido */

	"loader": {
		"name": "WEBAPP BUILDER",
		"version": "1.0.0",
		"release": true,
		"loading": 1,
		/* 7 opciones */

		"compatibilityFirst": "...",
		/* primer chequeo de compatibilidad, métodos nativos */

		"files": {
			"index": "index.html"
		},

		"folders": {...},
		/* configuración de folders */

		"viewport": "...",
		/* configuración del elemento meta: viewport */

		"contentSecurity": "...",
		/* configuración del elemento meta: content-security */

		"bower": {
			"platform": {
				"version": "1.3.0",
				"js-dev": ["platform/platform.js"],
				"js-prod": ["platform/platform.min.js"],
				"generate-js": true
			},
			...
		},
		/* componentes a usar (solo para _loader_), si no se quiere alguno de los por default, en el config del proyecto setearle null. También soporta la generación de los minificados si es que la librería no lo provee, usar "generate-js": true */

		"text": {...}
		/* todos los textos del loader (no soporta i18n) */
	}
}
```



CHANGELOG.md
18/01/16 v0.10.0
it was public on github! (01/01/06 00:02)
redo create.js with templates
englished!
doc updated
dependencies updated
continuous integration added


03/01/16 v0.9.0
Rehecho completamente el sistema de automatizacion  
TESTS! casi 200!  
Ahora soportar globs y muchas mejoras mas.  

15/10/13 v0.1.0  
Todo actualizado  
Proceso de gulp revisado y ajustado  

15/10/08 v0.1.0  
dependencias actualizadas  

15/04/18 v0.0.5  
mejoras sobre lo anterior  
bootstrap, jquery y swiper removido y pasado al projecto  

15/04/15 v0.0.4  
!reestructuracion completa!  
Injection dinamica de recursos segun stage con los app.json  
wrapeo de projecto  
muchismas cosas mas  

14/03/15 v0.0.2  
muchos cambios estructurales, integrando con app angular  
soporte a request sincronos y asincronos :)  

08/03/15  
varios fixes y loaders added  

23/02/15 v0.0.1  
Terminado, primera version  

01/02/15  
Recuperando files y codigo de projecto *** para reutilizar en este  


LOADER/www/config.js
///**
//* Created by Crystian on 10/18/2014.
//*/
//
//
////Gulp: Automatic by gulp (with install and build), don't CHANGE it manually or I'LL CUT YOUR HANDS!!!
//_loaderCfg.appName		= 'Number';//install
//_loaderCfg.loaderVersion= '0.0.1';//install
//_loaderCfg.gaId			= 'UA-XXXXXXXX-Z';//install
//_loaderCfg.appId		= 'com.languagesgym.number';//install
//_loaderCfg.appInstaller = 'com.languagesgym.installer';//install
//_loaderCfg.isCordovaDevice= false;//build
////END by gulp
//
//
////primer chequeo, si no es compatible con esto, se cancela el loader!
//_loaderCfg.compatibilityFirst = function(){
//	'use strict';
//
//	//jshint maxcomplexity:false
//	var arr = [];
//
//	//install is going to replace it:
//	//compatibilityFirst
//	return !!(window.document.querySelector &&
//	window.console &&window.document.addEventListener &&
//	(window.document.getElementsByTagName('body')[0]).classList &&
//	window.localStorage &&
//	window.navigator &&
//	window.XMLHttpRequest &&
//	window.applicationCache &&
//	window.document.createEvent &&
//	arr.map &&
//	arr.every);
//	//ENDcompatibilityFirst
//};
//
//
///*
// el primer elemento es incompatible 100%, menores e igual de
// el segundo es browser viejo, compatible, menores e igual de
// el tercero es compatible con ese y superiores, mayores e igual de
// */
////nombre raro intencional es sobre compatibilidad
//_loaderCfg.matrix = {
//	"ie": [
//		9.9,
//		10.9,
//		11
//	],
//	"firefox": [
//		23.9,
//		25.9,
//		26
//	],
//	"android": [
//		3.9,
//		3.9,
//		4
//	],
//	"chrome": [
//		17.9,
//		28.9,
//		29
//	],
//	"safari": [
//		6,
//		6,
//		7
//	],
//	"ios": [
//		6,
//		6,
//		7
//	]
//};//no borrar
//// 'operamini':[7	,7	,7],
//// 'bb':		[6	,7	,10],
//// 'ieMobile': [9	,9	,10]
//
//// TODO revisar opera, anda raro en la virtual con xp,
//// por otro lado los parametros que toma son de compatibles,
//// y por ende usa a chrome :S
//// 'opera':	[16	,17	,18]
//
////jshint maxlen:false
//_loaderCfg.i18n = {
//	loader : {//loader
//		incompatibleByFeatures : 'Navegador incompatible (no se encontraron features), por tu seguridad actualizalo o usa otro, ver en F.A.Q.',
//		incompatibleByDiag : 'Navegador incompatible, por tu seguridad actualizalo o usa otro, ver en F.A.Q.',
//		semiIncompatible : 'Navegador no reconocido o con problemas de compatibilidad, la aplicacion funcionara pero es recomendable actualizarlo o usar otro, ver en el F.A.Q. \nhttp://languagesgym.com/faq',
//		faqLink : '<a href="http://languagesgym.com/faq">http://languagesgym.com/faq</a>',
//		errorRequestFile : 'Error leyendo el archivo del servidor, por favor vuelva a intentar mas tarde',
//		errorTimeoutServer : 'Se agoto el tiempo de espera, verifica que tengas internet y este andando bien'
//	},
//
//	//this errors are lies, solo para enganiar al malisioso
//	consoleError: [
//		'Error unknown',
//		'Error about connection stablished',
//		'Error 404',
//		'Error connection lost, try again',
//		'Command unknown',
//		undefined,
//		'Null pointer exception',
//		'Error jquery plugin',
//		'Error ferrasca the boilerplate',
//		'Error map not found',
//		'Error in obliteration rast'
//	]
//
//};